### DISCUSSION TITLE
Need help with Regex (simple word-filter)

### DISCUSSION THREAD

**Discussion by pc-bastler** [rspamd/rspamd#4458] [Q&A]
Hi,

I want to create a simple word-filter (e.g. 'bit.ly', '*innen' (the Asterisk must not be interpreted as Wildcard), ':innen') for the body of an e-mail.
I know that this should be possible with the Regex-module, but I don't get it how to configure this.

Could you please assist me?

Best wishes

**Comment by ManChicken1911** [comment]
You could do this with a multimap. Regex is a deep subject, but first note that . and * and : need to be escaped with a backslash to match the actual characters as they are otherwise regex special characters. So:

`bit\.ly` would match bit.ly
`\*innen` would match the literal *innen
`\:innen` would match :innen

To make a multimap to look for such things in the body of a message, you might create/add to `/usr/local/etc/rspamd/local.d/multimap.conf` and do something like:
```
my_bad_content {
  type = "content";
  filter = "rawtext";
  map = "/usr/local/etc/rspamd/my_bad_content.txt";
  description = "Message contains bad content patterns";
  symbol = "MY_BAD_CONTENT";
  regexp = true;
  score = 5;
}
```

and `/usr/local/etc/rspamd/my_bad_content.txt` would contain:

```
/bit\.ly/
/\*innen/
/\:innen/
```

Note the lines are surrounded by / / - Also keep in mind that this will match those characters anywhere; for instance if the message said "This cool new site is called obit.lying!" it would match the bit.ly pattern.

Again regex is a whole thing unto its self but hopefully this should give you an idea of how to start.
